### PR TITLE
Install updates from Files

### LINKED ISSUE #1513: Ability of checking updates actively
**Describe the solution you'd like**
An app can check for updates actively and request silent downloads and installation, see 
https://docs.microsoft.com/en-us/windows/uwp/packaging/self-install-package-updates.
It can keep users who disabled Automatically Update in Microsoft Store up-to-date, and it can also inform users about the up-coming update contents.

I think it's helpful for improving user experience.

### PR DESCRIPTION
**Pull request by Thomas1664**
Fixes #1513 
I implemented the feature to install updates via a button in the about settings page.

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@Thomas1664 Instead of adding a button to check for updates, it should autmaticly check for udpates when launching the app.

**Comment by Thomas1664** [comment]
I didn't want to implement it yet because if the code is wrong the app crashes

**Comment by yaichenbaum** [comment]
> I didn't want to implement it yet because if the code is wrong the app crashes

That's fine, we will test it in a flight group to make sure it works correctly first, and if needed, we can wrap it in a try/catch.

**Comment by Jaiganeshkumaran** [comment]
> @Thomas1664 Instead of adding a button to check for updates, it should autmaticly check for udpates when launching the app.

No need. It will slow down the app startup. Also this app is available on the Store so it will check for updates automatically. Users will click the update button only if the store doesn't check,.

File: Files/Views/SettingsPages/About.xaml [hunk 1]
```diff
@@ -188,6 +188,8 @@
                             VerticalAlignment="Center"
                             Text="License:" />
                         <HyperlinkButton NavigateUri="https://github.com/QL-Win/QuickLook/blob/master/LICENSE-GPL.txt">GPL</HyperlinkButton>
+                        <Button Content="Check for updates" Click="UpdateCheck_Click">
```
Comment (yaichenbaum) [inline Files/Views/SettingsPages/About.xaml:191]: Please remove the button and have the code checking for updates run when the app is first launched.

**Review by yair100 — CHANGES_REQUESTED** [review]

File: Files/Views/SettingsPages/About.xaml (hunk 1 above)
Comment (tsvietOK) [reply Files/Views/SettingsPages/About.xaml]: @yair100 I don't think it is a good idea to show update dialog on every app launch.

Comment (yaichenbaum) [reply Files/Views/SettingsPages/About.xaml:191]: There is no need to show a dialog every time the app is opened, the only time a dialog will be displayed is if a mandatory update is detected.

Comment (boscorelly) [reply Files/Views/SettingsPages/About.xaml:191]: i don't think it must be at startup by default. Maybe an option. Noting more, as store is already updating apps

Comment (yaichenbaum) [reply Files/Views/SettingsPages/About.xaml:191]: @boscorelly This pull request is regarding requiring mandatory updates.

**Comment by yaichenbaum** [comment]
image
@Thomas1664 Is this because the app is not connected to the Store in our debug builds?

**Comment by Thomas1664** [comment]
Yes

**Review by yair100 — DISMISSED** [review]

**Review by yair100 — DISMISSED** [review]

**Review by yair100 — APPROVED** [review]

**Review by tsvietOK — APPROVED** [review]

**Comment by myokeeh** [comment]
> Yes

How did you get around this?

**Comment by Thomas1664** [comment]
> > Yes
> 
> How did you get around this?

See the attatched issue. We used a flight group.

**Comment by myokeeh** [comment]
> See the attached issue. We used a flight group.

interesting. i've been struggling with this. I have a flight group also, but this still happens.

**Comment by yaichenbaum** [comment]
@myokeeh We ended up wrapping this in a try/catch, for some reason we still get the error every time and updates are not detected, but at least the app isn't crashing.

**Comment by myokeeh** [comment]
@yair100, so `GetAppAndOptionalStorePackageUpdatesAsync` is in fact still broken--no updates detected. 🤷‍♂️I hope somebody at Microsoft gets on this and get it fixed.

**Pull request merged**